tutorial3_distributions/demo/main.js

```diff
@@ -1,133 +1,127 @@
-   * pull out all the state values from the data create a `Set` out of them
-   * This is useful for creating a list of just *unique* values.
-   * The `[...Set]` syntax 'unpacks' the set into an array so that we can use it in our `data()` method.
-   * To see what this does, try the following in your browser console:
-   *    new Set([1, 2, 1, 4, 5, 6, 4, 6, 10])
-   * and:
-   *    [...new Set([1, 2, 1, 4, 5, 6, 4, 6, 10])]
-   */
-  const uniqueStates = [...new Set(data.map(d => d.state))];
-  // add dropdown for interaction
-  // select reference: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select
+  // add dropdown (HTML selection) for interaction
+  // HTML select reference: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select
   const selectElement = d3.select("#dropdown").on("change", function() {
-    console.log("new selected state is", this.value);
+    console.log("new selected party is", this.value);
     // `this` === the selectElement
     // this.value holds the dropdown value a user just selected
-    selectedState = this.value;
+    state.selectedParty = this.value;
     draw(); // re-draw the graph based on this new selection
   });
 
   // add in dropdown values from the unique options in the data
   selectElement
     .selectAll("option")
-    .data(uniqueStates)
+    .data(["D", "R", "Both"])
     .join("option")
     .attr("value", d => d)
-    .text(d => d);
+    .text(d => d)
 
   // create an svg element in our main `d3-container` element
   svg = d3
     .select("#d3-container")
     .append("svg")
-    .attr("width", width)
-    .attr("height", height);
+    .attr("width", constants.width)
+    .attr("height", constants.height);
 
   // add the xAxis
   svg
     .append("g")
     .attr("class", "x-axis")
-    .attr("transform", `translate(0,${height - margin.bottom})`)
+    .attr("transform", `translate(0,${constants.height - constants.margin.bottom})`)
     .call(xAxis);
 
   // add the xAxis
   svg
     .append("g")
     .attr("class", "y-axis")
-    .attr("transform", `translate(${margin.left},0)`)
+    .attr("transform", `translate(${constants.margin.left},0)`)
     .call(yAxis);
 
   draw();
 }
 
+// DRAW FUNCTION
 function draw() {
   // we call this everytime there is an update to the data/state
 
-  // filter the data for the selectedState
-  let filteredData = data;
-  // if there is a selectedState, filter the data before drawing it
-  if (selectedState) {
-    filteredData = data.filter(d => d.state === selectedState);
+  // filter the data for the selectedParty
+  let filteredData = state.data;
+  // if there is a selectedParty, filter the data before drawing it
+  if (state.selectedParty) {
+    filteredData = state.data.filter(d => 
+      state.selectedParty === "Both" 
+    || d.party === state.selectedParty);
+    console.log("new filtered data includes", filteredData.length, "politicians");
```

Comment:
this is a bit complicated (although well written) and we might not need it (since there will never be a situation in which a party isn't selected) but they might need it for other applications